app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsEmailAddressController.scala

```diff
@@ -77,11 +96,87 @@ class ContactDetailsEmailAddressController @Inject() (
 
   private def updateRegistration(
     formData: EmailAddress
-  )(implicit req: JourneyRequest[AnyContent]): Future[Either[ServiceError, Registration]] =
-    update { registration =>
-      val updatedEmailAddress =
-        registration.primaryContactDetails.copy(email = Some(formData.value))
-      registration.copy(primaryContactDetails = updatedEmailAddress)
+  )(implicit request: JourneyRequest[AnyContent]): Future[Either[ServiceError, Registration]] =
+    request.user.identityData.credentials.map { creds =>
+      emailVerificationConnector.getStatus(creds.providerId).flatMap {
+        case Right(emailStatusResponse) =>
+          emailStatusResponse match {
+            case Some(response) =>
+              update { registration =>
+                registration.copy(primaryContactDetails =
+                                    updatedPrimaryContactDetails(formData, registration),
+                                  metaData = registration.metaData.add(response.emails)
+                )
+              }
+            case None =>
+              update { registration =>
+                registration.copy(primaryContactDetails =
+                  updatedPrimaryContactDetails(formData, registration)
+                )
```

Comment:
We need to a test to cover this journey.

Reply:
Done